src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -209,7 +209,7 @@ impl UrlFragment {
             &UrlFragment::Item(def_id) => {
                 let kind = match tcx.def_kind(def_id) {
                     DefKind::AssocFn => {
-                        if tcx.defaultness(def_id).has_value() {
+                        if tcx.associated_item(def_id).defaultness(tcx).has_value() {
```

Comment:
What is the difference?

Reply:
This version protects against querying defaultness on inherent impl items since now that will ICE.